mysql-test/suite/galera/t/galera_garbd_backup.test

```diff
@@ -0,0 +1,65 @@
+#
+# Test a single node with an arbitrator.  Startup a 2-node cluster, shut down node 2, and use it's ports
+# for the garbd.
+#
+
+--source include/galera_cluster.inc
+
+SHOW STATUS LIKE 'wsrep_flow_control_interval';
+
+--connection node_1
+CALL mtr.add_suppression("Protocol violation\. JOIN message sender 1\.0 \(.*\) is not in state transfer \(SYNCED\)");
+CALL mtr.add_suppression("State transfer to .*");
+CALL mtr.add_suppression("Process completed with error:.*");
+CALL mtr.add_suppression("Command did not run:.*");
+CALL mtr.add_suppression("WSREP-SST.*");
+
+--connection node_2
+CALL mtr.add_suppression("Protocol violation\. JOIN message sender 1\.0 \(.*\) is not in state transfer \(SYNCED\)");
+CALL mtr.add_suppression("State transfer to .*");
+CALL mtr.add_suppression("Process completed with error:.*");
+CALL mtr.add_suppression("Command did not run:.*");
+CALL mtr.add_suppression("WSREP-SST.*");
+
+--echo Starting garbd ...
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --recv-script="sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+--sleep 2 
+
+# Error: incorrect donor sst script
+--error 1
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v3:127.0.0.1:9998/xtrabackup_sst//1" --recv-script="sst_garbd_test 9998" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd2.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+--sleep 2 
+
+# Error: incorrect receiver script
+--error 2
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --recv-script="not_sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd2.log 2>&1 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+#--echo Restarting node #2 to satisfy MTR's end-of-test checks
+#--connection node_2
+#--source include/start_mysqld.inc
+
+# Simulate two connection error 
+
+--exec `dirname $WSREP_PROVIDER`/../bin/garbd --sst="xtrabackup-v2:127.0.0.1:9999/xtrabackup_sst//1" --donor node1 --recv-script="sst_garbd_test 9999" --address "gcomm://127.0.0.1:$NODE_GALERAPORT_1" --group my_wsrep_cluster --options 'evs.inactive_timeout=PT6S;base_port=$NODE_GALERAPORT_3' > /tmp/garbd3.log 2>&1 & 
+SELECT VARIABLE_VALUE FROM performance_schema.global_status WHERE VARIABLE_NAME = 'wsrep_cluster_size';
+
+# Depending on how much time it passes since the previous garbd start, this either fails or succeeds.
```

Comment:
How do we know if we are testing problematic situation and not always following happy path when both garbd receive the backup? 
Is it possible to pause server on some debug_sync point to force problematic situation?

Reply:
It's not possible: the issue happens in garbd, which doesn't have debug sync functionality.